pootle/apps/pootle_app/management/commands/__init__.py

```diff
@@ -141,20 +142,21 @@ def handle_all(self, **options):
         if options["no_rq"]:
             set_sync_mode(options['noinput'])
 
-        if self.process_disabled_projects:
-            project_query = Project.objects.all()
-        else:
-            project_query = Project.objects.enabled()
+        tps = TranslationProject.objects.live().order_by(
+            "project__code", "language__code").select_related(
+                "project",
+                "language",
+                "directory",
+                "project__source_language")
 
-        if self.projects:
-            project_query = project_query.filter(code__in=self.projects)
+        if not self.process_disabled_projects:
+            tps = tps.exclude(project__disabled=True)
 
-        for project in project_query.iterator():
-            tp_query = project.translationproject_set.live() \
-                              .order_by('language__code')
+        if self.projects:
+            tps = tps.filter(project__code__in=self.projects)
 
-            if self.languages:
-                tp_query = tp_query.filter(language__code__in=self.languages)
+        if self.languages:
+            tps = tps.filter(language__code__in=self.languages)
 
-            for tp in tp_query.iterator():
-                self.do_translation_project(tp, **options)
+        for tp in tps.iterator():
```

Comment:
It is a personal nit, but I would sort the TPs in some way, so when we run the commands we get the TPs processed in a predictable order, for example process the TPs for a given project first, and then for the next one...